src/riffvideo.cpp

```diff
@@ -527,660 +376,382 @@ void RiffVideo::readMetadata() {
-  uint64_t cur_pos = io_->tell();
-  const size_t bufMinSize = 100;
-  DataBuf buf(bufMinSize);
-  io_->read(buf.data(), size);
-  if (!i)
-    xmpData_["Xmp.video.DateUTC"] = buf.data();
-  else
-    xmpData_["Xmp.video.StreamName"] = buf.data();
-  io_->seek(cur_pos + size, BasicIo::beg);
-}  // RiffVideo::dateTimeOriginal
-
-void RiffVideo::odmlTagsHandler() {
-  const size_t bufMinSize = 100;
-  DataBuf buf(bufMinSize);
-  io_->seek(-12, BasicIo::cur);
-  io_->read(buf.data(), RIFF_TAG_SIZE);
-  size_t size = Exiv2::getULong(buf.data(), littleEndian);
-  size_t size2 = size;
-
-  uint64_t cur_pos = io_->tell();
-  io_->read(buf.data(), RIFF_TAG_SIZE);
-  size -= RIFF_TAG_SIZE;
-
-  while (size > 0) {
-    io_->read(buf.data(), RIFF_TAG_SIZE);
-    size -= RIFF_TAG_SIZE;
-    if (equalsRiffTag(buf, "DMLH")) {
-      io_->read(buf.data(), RIFF_TAG_SIZE);
-      size -= RIFF_TAG_SIZE;
-      io_->read(buf.data(), RIFF_TAG_SIZE);
-      size -= RIFF_TAG_SIZE;
-      xmpData_["Xmp.video.TotalFrameCount"] = Exiv2::getULong(buf.data(), littleEndian);
-    }
-  }
-  io_->seek(cur_pos + size2, BasicIo::beg);
-}  // RiffVideo::odmlTagsHandler
-
-void RiffVideo::skipListData() {
-  DataBuf buf(RIFF_TAG_SIZE + 1);
-  io_->seek(-12, BasicIo::cur);
-  io_->read(buf.data(), RIFF_TAG_SIZE);
-  size_t size = Exiv2::getULong(buf.data(), littleEndian);
-
-  uint64_t cur_pos = io_->tell();
-  io_->seek(cur_pos + size, BasicIo::beg);
-}  // RiffVideo::skipListData
-
-void RiffVideo::copyTagValue(DataBuf& buf_dest, DataBuf& buf_src, size_t index) {
-  buf_dest.data()[0] = buf_src.data()[0 + index];
-  buf_dest.data()[1] = buf_src.data()[1 + index];
-  buf_dest.data()[2] = buf_src.data()[2 + index];
-  buf_dest.data()[3] = buf_src.data()[3 + index];
-}
+void RiffVideo::readAviHeader() {
+#ifdef EXIV2_DEBUG_MESSAGES
+  EXV_DEBUG << "--> dwMicroSecPerFrame    = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwMaxBytesPerSec      = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwPaddingGranularity  = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwFlags               = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwTotalFrames         = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwInitialFrames       = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwStreams             = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwSuggestedBufferSize = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwWidth               = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwHeight              = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwReserved1           = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwReserved2           = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwReserved3           = " << readDWORDTag(io_) << std::endl;
+  EXV_DEBUG << "--> dwReserved4           = " << readDWORDTag(io_) << std::endl;
+#else
```

Comment:
```suggestion
#endif
```

I don't think the code below this should be commented out when `EXIV2_DEBUG_MESSAGES` is on, should it?

If you agree with this suggestion, then there are several other places in this file that also need to be fixed.

Reply:
Thank you @kevinbackhouse ,
Done  here [db14131](https://github.com/Exiv2/exiv2/pull/2458/commits/db1413126237bc8cc81abd6a583e4e9782f3e2aa)